src/racer/core.rs

```diff
@@ -648,17 +666,29 @@ pub fn complete_from_file(src: &str, filepath: &path::Path,
             let context = ast::get_type_of(contextstr.to_owned(), filepath, pos, session);
             debug!("complete_from_file context is {:?}", context);
             context.map(|ty| {
-                if let Ty::TyMatch(m) = ty {
-                    for m in nameres::search_for_field_or_method(m, searchstr, SearchType::StartsWith, session) {
-                        out.push(m)
-                    }
-                }
+                complete_field_for_ty(ty, searchstr, SearchType::StartsWith, session, &mut out);
             });
         }
     }
     out.into_iter()
 }
 
+fn complete_field_for_ty(ty: Ty, searchstr: &str, stype: SearchType, session: &Session, out: &mut Vec<Match>) {
+    // TODO would be nice if this and other methods could operate on a ref instead of requiring
```

Comment:
We may at some point want to handle `String` and `Vec` here (perhaps a special case?), too, since they get a lot of methods out of autoderef through slices.